### PR TITLE
Solve conflicts due to lmfit new v0.9

### PR DESCRIPTION
**Pull request by licode**
still in process

### CODE REVIEW

**Comment by ericdill** [comment]
Travis is testing against lmfit 0.8.3. Did you mean to test against 0.9.0?

**Comment by licode** [comment]
Yes, let's use 0.9.0 in Travis. But I would suggest we still use 0.8.3 for installation until I fully understand the conflicts.

**Comment by licode** [comment]
There are lots of changes in this new version. I already spent several hrs yesterday.

**Comment by ericdill** [comment]
Alright, I'll need to build the v0.9.0 tag and upload it to
anaconda.org/lightsource2. Can you update .Travis.yml to have lmfit=0.9.0?

**Comment by sameera2004** [comment]
@Li I agree there are lot of changes in 0.9.0 previously I installed it. 
When I try to run tests in skxray and got lot of lmfit errors, were able to get rid of those with lmfit 0.8.3.

**Comment by ericdill** [comment]
Alright, I just built lmfit 0.9.0 and uploaded it to lightsource2 and scikit-xray. I'm restarting travis to make sure it passes with the tagged release. The previous pass that you got was on 0.9.0rc2.

File: skxray/core/fitting/xrf_model.py [hunk 1]
```diff
@@ -1025,10 +1025,14 @@ def construct_linear_model(channel_number, params,
...
     for elemental_line in elemental_lines:
+        #print(elemental_line)
         e_model = MS.setup_element_model(elemental_line,
                                          default_area=default_area)
+
+        #MS.assemble_models()
         if e_model:
-            p = e_model.make_params()
+            new_model = MS.compton + MS.elastic + e_model
+            p = new_model.make_params()
```
Comment (tacaswell) [inline skxray/core/fitting/xrf_model.py]: This looks like it is science logic changes, not compatibility changes?

File: skxray/core/fitting/xrf_model.py [hunk 2]
```diff
@@ -1043,7 +1047,8 @@ def construct_linear_model(channel_number, params,
     element_area.update({'compton': p['compton_amplitude'].value})
     selected_elements.append('compton')
 
-    p = MS.elastic.make_params()
+    compton_elastic = MS.compton + MS.elastic
+    p = compton_elastic.make_params()
```
Comment (tacaswell) [inline skxray/core/fitting/xrf_model.py]: Same for this

File: skxray/core/fitting/xrf_model.py (hunk 1 above)
Comment (licode) [inline skxray/core/fitting/xrf_model.py]: No, this is due to change of new evaluate method on parameter in v0.9. 

Let's ignore this pr, until the final version of 0.9 is determined. I think it is still debatable on several important points.

Comment (danielballan) [inline skxray/core/fitting/xrf_model.py]: Yeah, I think a v0.9.1 will be coming soon to roll back some accidental API breakages.

Comment (tacaswell) [inline skxray/core/fitting/xrf_model.py]: can someone explain to me what changed and how this is supposed to work?

Comment (licode) [inline skxray/core/fitting/xrf_model.py]: Will do that during the weekend. I am busy at beamline today.

**Comment by tacaswell** [comment]
@licode Status of this?

**Comment by licode** [comment]
I will close this issue for now. I will send a new pr after fully working out the issues in lmfit.

**Comment by tacaswell** [comment]
Did you follow up on this?

**Comment by licode** [comment]
Thanks for reminding me. I am going to work on this again.

**Comment by tacaswell** [comment]
Could you have an answer for this by Monday or Tuesday of next week.  We need to decide which version of lmfit to install on all of the beamlines for this run.

**Comment by licode** [comment]
Let's stay with 0.8.3 for now.

**Pull request closed without merging**